From 6a47e02b437ae0a0d3d177b6f7e1c8a3562772af Mon Sep 17 00:00:00 2001
From: Saggi Mizrahi <smizrahi@redhat.com>
Date: Wed, 27 Apr 2011 11:13:42 +0300
Subject: [PATCH 2/3] Always use uncached tags when reading metadata

When we read metadata we assume something happend where the current host
was not involved. Refreshing the VG is recommended because lvm chache is
not cluster aware

Change-Id: I0f8187c039cfa0b4f3a17bc9a875c34a292946df
Reviewed-on: http://gerrit.usersys.redhat.com/327
Tested-by: Dan Kenigsberg <danken@redhat.com>
Reviewed-by: Dan Kenigsberg <danken@redhat.com>
---
 vdsm/storage/blockSD.py |    1 +
 vdsm/storage/spm.py     |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/vdsm/storage/blockSD.py b/vdsm/storage/blockSD.py
index e283461..420a2d1 100644
--- a/vdsm/storage/blockSD.py
+++ b/vdsm/storage/blockSD.py
@@ -106,6 +106,7 @@ class VGTagMetadataRW(object):
         self._vgName = vgName
 
     def readlines(self):
+        lvm.refreshVG(self._vgName)
         vg = lvm.getVG(self._vgName)
         metadata = []
         for tag in vg.tags:
diff --git a/vdsm/storage/spm.py b/vdsm/storage/spm.py
index c24114f..7e98dd4 100644
--- a/vdsm/storage/spm.py
+++ b/vdsm/storage/spm.py
@@ -517,6 +517,7 @@ class SPM:
                 self.lver = int(oldlver) + 1
                 self.pools[spUUID] = pool
 
+                pool.invalidateMetadata()
                 pool.setMetaParams({sp.PMDK_LVER: self.lver,
                     sp.PMDK_SPM_ID: pool.id})
                 self.pools[spUUID]._maxHostID = maxHostID
-- 
1.7.3.4

